src/ir/ctx.rs

```diff
@@ -1,21 +1,27 @@
 use super::Component;
-use crate::utils::Idx;
+use crate::utils::{Idx, IdxLike};
 
 /// A context for storing values with their indices.
 /// The context is indexed by [`Idx<T>`].
-pub trait Ctx<T> {
+pub trait Ctx<T, K = Idx<T>>
+where
+    K: IdxLike<T>,
```

Comment:
The `IdxLike` stuff is also implemented in #288 in pretty much the exact same way so merging this will likely cause conflicts with that (no changes needed, just a note).

Reply:
Thanks!